src/libsyntax/ext/base.rs

```diff
@@ -546,6 +557,7 @@ pub struct ExtCtxt<'a> {
     pub ecfg: expand::ExpansionConfig<'a>,
     pub crate_root: Option<&'static str>,
     pub feature_gated_cfgs: &'a mut Vec<GatedCfgAttr>,
+    pub loader: &'a mut MacroLoader,
```

Comment:
My main reason for using a trait object is that `macro_import::MacroLoader` is defined and implemented in `librustc_metadata`, so `libsyntax` can't use it directly.